### PR TITLE
Die if uinput module is neither loaded nor built into the kernel

### PR DESCRIPTION
**Pull request by fauxmight**

### CODE REVIEW

**Comment by fauxmight** [comment]
Holy frijoles, Batman.
Cook booking Rust when you do not know Rust is a bear.

In my testing, this works correctly, dies when it should, and announces accurately that uinput is either a loaded module or is built into the kernel.

HOWEVER, I strongly suspect my completely ignorant coding style in Rust will look to those who know Rust ... like completely ignorant coding style in Rust.

I do not ask you developers to "go easy on me; it's my first try at this," but rather to be brutal and instructive. I have a thick skin and would rather learn than be treated gently.

It took me about seven hours of learning Rust notions to make something that would compile ... and another two to make something that would do what it was intended to do. This does not mean that what little I've written is stylistically desirable Rust. I come from Java (despite hating Java) and my only coding pal (we're both dentists, not professional developers, though he is a significantly better coder than am I) regularly notes that whether I write Haskell, Python or C, it all looks like Java.

File: src/main.rs [hunk 1]
```diff
@@ -267,7 +267,32 @@ impl EventObserver for KeyLogger {
...
+	let modules = std::fs::read_to_string("/proc/modules").unwrap();
+	for line in modules.lines() {
+		if line.starts_with("uinput ") {
+			info!("'uinput' module is loaded");
+			return true;
+		}
+	}
+	return false;
+}
+
```
Comment (samvel1024) [inline src/main.rs:291]: Thanks for the PR

How confident are you about this being cross-platform? Will it work across most popular linux desktop distros/kernel versions?

Comment (fauxmight) [reply src/main.rs:291]: I am confident only that it compiles cleanly on Arch. Your question is very fair. I have not tested it on other Linux distros/kernels.

Comment (samvel1024) [reply src/main.rs:291]: compilation is of course not the concern. Main concern is runtime failure (based on distro, kernel version etc..). I think this should be handled in a wrapper script.

Comment (fauxmight) [reply src/main.rs:291]: Do you mean that you want a wrapper script in the `kbct` codebase or that a wrapper script should be added by distros when packaging `kbct`?

Comment (samvel1024) [reply src/main.rs:291]: By distros, for example for systemd you can add pre-init script that will load kernel module.

**Comment by fauxmight** [comment]
@samvel1024 Understood.

For posterity: I created this PR because when this fails, it fails silently. `kbct` runs and appears to be functioning, but is completely non-operational and indicates no error or issue. Yes, the need for the `uinput` module was added to documentation (PR #26), but when this program runs and `uinput` is not available in the kernel or loaded as a module, it looks perfectly operational while nothing will actually work as expected.

There is absolutely *no reason* that an individual handling packaging of `kbct` for a distribution would intuit this and just *know* to add a wrapper script. Furthermore, anyone handling the packaging of `kbct` will have to go source diving to discover that this is the reason it is failing (when it looks like it *isn't* failing).

Keeping `kbct` the way it is without this or some other check/warning for the missing module seems (as discussed in issue #25) counterproductive.

That being said, it is your code and your project. Your decision is respected, and I am closing the PR.

**Pull request closed without merging**